src/lib/optimizer/strategy/chunk_pruning_rule.cpp

```diff
@@ -106,4 +105,8 @@ std::set<ChunkID> ChunkPruningRule::_compute_exclude_list(
   return result;
 }
 
+bool ChunkPruningRule::_is_non_filtering_node(const AbstractLQPNode& node) const {
```

Comment:
Is that something we can recycle somewhere else?

Reply:
The rule of thumb that I've grown to like is that if something is used three times it's okay to generalize it, everything below that is overengineering.